context/context.go

```diff
@@ -202,3 +255,82 @@ func (c *ctx) returnToPool() {
 	c.Reset()
 	c.pool.Put(c)
 }
+
+func (c *ctx) newChildContext() Context {
+	var child Context
+	if c.pool == nil {
+		child = NewContext()
+	} else {
+		child = c.pool.Get()
+	}
+
+	// is there a better way to do this?
```

Comment:
Hm, maybe try to cast it and if it fails then create one yourself?  Maybe we should create the following two methods so there is always one way to get a concrete `ctx`.

i.e. something like this:
```go

func NewContext() Context {
  return newContext()
}

func newContext() *ctx {
  return &ctx{
    //.. whatever
  }
}

// ..

func (c *ctx) newChild() Context {
	var child *ctx
	if c.pool != nil {
		pooled, ok := c.pool.Get().(*ctx)
		if ok {
			child = pooled
		}
	}

	if child == nil {
		child = newContext() // which returns *ctx
	}

	child.setParentCtx(c)
	return child
}

```